libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -691,7 +699,33 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		for (size_t i = 0; i < arguments.size(); ++i)
 			args += convert(*arguments[i], *parameterTypes[i]).stackSlots();
 
-		if (functionDef)
+		if (_functionCall.annotation().kind == FunctionCallKind::StructConstructorCall)
+		{
+			TypeType const& type = dynamic_cast<TypeType const&>(*_functionCall.expression().annotation().type);
+			auto const& structType = dynamic_cast<StructType const&>(*type.actualType());
+
+			string const structVar = m_context.newYulVariable();
+			vector<map<string, string>> members(args.size());
+			for (size_t i = 0; i < members.size(); i++)
+				members[i]["var"] = args[i];
+
+			m_code << Whiskers(R"(
+				let <struct> := <structAlloc>()
+				let <offset> := <struct>
+				<#member>
+					mstore(<offset>, <var>)
```

Comment:
Maybe it is better to use an actual loop here and extend `m_code`?

Reply:
Ah, actually calling `mstore` is not enough here - some types need a special routine to write them to memory.

Reply:
`writeToMemoryFunction`